packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -368,6 +368,39 @@ abstract final class SystemChrome {
   ///
   /// ## Limitations
   ///
+  /// ### Android
+  ///
+  /// Android screens may choose to [letterbox](https://developer.android.com/guide/practices/enhanced-letterboxing)
+  /// applications that lock orientation, particularly on larger screens. When
+  /// letterboxing occurs on Android, the [MediaQueryData.size] reports the
+  /// letterboxed size, not the full screen size. Applications that make
+  /// decisions about whether to lock orientation based on the screen size
+  /// must use the `display` property of the current [FlutterView].
+  ///
+  /// ```dart
+  /// // Lock the screen to portrait if it is less than 600 logical pixels wide.
+  /// const double kOrientationLockBreakpoint = 600;
+  /// class OrientationLockingObserver extends WidgetsBindingObserver {
+  ///   @override
+  ///   void didChangeMetrics() {
+  ///     final ui.Display display = PlatformDispatcher.instance.implicitView!.display;
```

Comment:
This is making a single view assumption that we worked so hard on to remove from the framework. What if `implicitView` is null? Any way we can make this look up the actual view things are getting rendered into? From a widget tree you can use `View.of`, for example.

Reply:
You don't have a widget tree in an observer, and right now the media query doesn't give you access to the display. I was going to hold off on that until it's implemented on at least Windows (right now it's only on Android, iOS, and macOS).

Are we expecting multi-window on Android to be a thing?

Reply:
> You don't have a widget tree in an observer,

An observer can be anything, including something that is in the widget tree, e.g. a State object.

> Are we expecting multi-window on Android to be a thing?

Multi-**view** will be a thing on Android, e.g. in add-to-app scenarios.

Reply:
Ok, I've updated the sample code to make a stateful widget that listens to the right view now, and happened to notice a bug in another semi-related sample which I fixed up.